wagtail/documents/templates/wagtaildocs/documents/list.html

```diff
@@ -36,11 +36,12 @@
     </thead>
     <tbody>
         {% trans "Select document" as checkbox_aria_label %}
+        {% querystring as querystring %}
```

Comment:
While this does work for now, I'm not sure how future-proof it is... do we really want the every GET param to be carried over? Looking at this now, I think it would be simpler and safer just to use pick up `collection_id` from the edited object. We wouldn't need to preserve GET params in the edit form that way, either. 

What do you think @bmihelac?

Reply:
The idea behind passing querystring is to support other filters as well (current and future) . For example if I search for document "test" and edit one document, I would be redirected to filtered documents after editing. If I remember this is what Django admin does.

Reply:
@bmihelac I see what you mean. Though, I probably should have highlighted that we already have a pattern used elsewhere, where a `?next=` value is passed in the initial GET request, then is included in a hidden form field, preserving the value in subsequent POST requests. As a Django aficionado, I'm sure you'll recognise this :) it's ever-so-slightly tidier, as it keeps data out of URLs, and can also be useful for other things (e.g. modeladmin listing button include a next value in the URL, so that users come back to the same listing when they are done).

Reply:
It looks like someone else has implemented this fix separately in https://github.com/wagtail/wagtail/pull/7810, so maybe we can simplify this PR somewhat?